services/proxy/pkg/middleware/policies_test.go

```diff
@@ -0,0 +1,229 @@
+package middleware_test
```

Comment:
These test are very complex because they test e2e functionality. Can't we refactor `Policies` func so we can better unit test it? If we could separate the request/response writing from the actual logic, we would only have to test the latter.

Reply:
uses mocks now